### PR TITLE
Added ChangeDescription to the FeatureChange GeoJSON serialization

### PR DESCRIPTION
**Pull request by lucaspcram**
### Description:
`FeatureChange` GeoJSON serialization now includes the `ChangeDescription` in an object called `description` under the `properties`.

Here is an example of a GeoJSON serialized `FeatureChange` that is updating an `Edge`. Check the `description` field in the `properties` object for all the juicy details:
```
{
  "type": "Feature",
  "bbox": [
    -122.052138,
    37.317585,
    -122.009566,
    37.390535
  ],
  "geometry": {
    "type": "LineString",
    "coordinates": [
      [
        -122.009566,
        37.33531
      ],
      [
        -122.031007,
        37.390535
      ],
      [
        -122.028932,
        37.332451
      ],
      [
        -122.052138,
        37.317585
      ],
      [
        -122.0304871,
        37.3314171
      ]
    ]
  },
  "properties": {
    "featureChangeType": "ADD",
    "meta-data": {},
    "description": {
      "type": "UPDATE",
      "descriptors": [
        {
          "name": "PARENT_RELATION",
          "type": "ADD",
          "afterView": "3"
        },
        {
          "name": "PARENT_RELATION",
          "type": "REMOVE",
          "beforeView": "1"
        },
        {
          "name": "START_NODE",
          "type": "UPDATE",
          "beforeView": "1",
          "afterView": "10"
        },
        {
          "name": "END_NODE",
          "type": "UPDATE",
          "beforeView": "2",
          "afterView": "20"
        },
        {
          "name": "GEOMETRY",
          "type": "REMOVE",
          "position": "0/5",
          "beforeView": "LINESTRING (-122.052138 37.317585, -122.0304871 37.3314171, -122.028932 37.332451)"
        },
        {
          "name": "GEOMETRY",
          "type": "ADD",
          "position": "5/5",
          "afterView": "LINESTRING (-122.028932 37.332451, -122.052138 37.317585, -122.0304871 37.3314171)"
        },
        {
          "name": "TAG",
          "type": "ADD",
          "key": "c",
          "value": "3"
        },
        {
          "name": "TAG",
          "type": "UPDATE",
          "key": "b",
          "value": "2a",
          "originalValue": "2"
        },
        {
          "name": "TAG",
          "type": "REMOVE",
          "key": "a",
          "value": "1"
        }
      ]
    },
    "entityType": "EDGE",
    "class": "org.openstreetmap.atlas.geography.atlas.complete.CompleteEdge",
    "identifier": 123,
    "tags": {
      "b": "2a",
      "c": "3"
    },
    "relations": [
      2,
      3
    ],
    "startNode": "10",
    "endNode": "20",
    "WKT": "LINESTRING (-122.009566 37.33531, -122.031007 37.390535, -122.028932 37.332451, -122.052138 37.317585, -122.0304871 37.3314171)",
    "bboxWKT": "POLYGON ((-122.052138 37.317585, -122.052138 37.390535, -122.009566 37.390535, -122.009566 37.317585, -122.052138 37.317585))"
  }
}
```

### Potential Impact:
N/A

### Unit Test Approach:
Added some additional unit tests to verify correct serialization.

### Test Results:
pass

------

In doubt: [Contributing Guidelines](CONTRIBUTING.md)

### CODE REVIEW

**Comment by matthieun** [comment]
For the Geometry updates, can we only display the locations that are updated/deleted? In case of large features with small changes, that would mean a really large output for nothing...

**Comment by lucaspcram** [comment]
@yazad3 @matthieun I've addressed your comments!

File: src/main/java/org/openstreetmap/atlas/geography/atlas/change/Change.java [hunk 1]
```diff
@@ -238,11 +239,30 @@ public void save(final WritableResource resource)
         new ChangeGeoJsonSerializer().accept(this, resource);
     }
 
+    /**
+     * Save a JSON representation of that feature change.
```
Comment (matthieun) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/change/Change.java]:
```suggestion
     * Save a JSON representation of that change.
```

File: src/main/java/org/openstreetmap/atlas/geography/atlas/change/description/descriptors/ChangeDescriptorComparator.java [hunk 2]
```diff
@@ -53,23 +54,72 @@ private int complexCompare(final ChangeDescriptor left, final ChangeDescriptor r
     private int genericSetChangeCompare(final GenericElementChangeDescriptor left,
             final GenericElementChangeDescriptor right)
     {
-        if (!left.getDescription().equals(right.getDescription()))
+        if (!left.getName().equals(right.getName()))
```
Comment (matthieun) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/change/description/descriptors/ChangeDescriptorComparator.java]: Here the name is an enum like above, right?
```suggestion
        if (left.getName() != right.getName())
```

File: src/main/java/org/openstreetmap/atlas/geography/atlas/change/description/descriptors/ChangeDescriptorComparator.java [hunk 3]
```diff
@@ -53,23 +54,72 @@ private int complexCompare(final ChangeDescriptor left, final ChangeDescriptor r
...
+        final Comparable rightAfterComparable = (Comparable) right.getAfterElement();
+        if (leftAfterComparable != null && rightAfterComparable != null
+                && !leftAfterComparable.equals(rightAfterComparable))
+        {
+            return leftAfterComparable.compareTo(rightAfterComparable);
+        }
+        else
+        {
+            /*
+             * If we start seeing this message in production, then that means either this comparison
```
Comment (matthieun) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/change/description/descriptors/ChangeDescriptorComparator.java]: How about we rephrase that to not be at the first person? Seems weird in code

File: src/main/java/org/openstreetmap/atlas/geography/atlas/change/description/descriptors/ChangeDescriptorComparator.java [hunk 4]
```diff
@@ -53,23 +54,72 @@ private int complexCompare(final ChangeDescriptor left, final ChangeDescriptor r
...
+        final Optional<String> leftAfterViewWkt = left.getAfterViewWkt();
+        final Optional<String> rightAfterViewWkt = right.getAfterViewWkt();
+        if (leftAfterViewWkt.isPresent() && rightAfterViewWkt.isPresent())
+        {
+            return leftAfterViewWkt.get().compareTo(rightAfterViewWkt.get());
+        }
+        else
+        {
+            /*
+             * If we start seeing this message in production, then that means either this comparison
```
Comment (matthieun) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/change/description/descriptors/ChangeDescriptorComparator.java]: Same suggestion

File: src/main/java/org/openstreetmap/atlas/geography/atlas/change/description/descriptors/GeometryChangeDescriptor.java [hunk 5]
```diff
@@ -21,6 +25,8 @@
  */
 public final class GeometryChangeDescriptor implements ChangeDescriptor
 {
+    private static final int TRUNCATE_WIDTH = 200;
```
Comment (matthieun) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/change/description/descriptors/GeometryChangeDescriptor.java]: 👍

File: src/test/java/org/openstreetmap/atlas/geography/atlas/change/FeatureChangeUnitTestFactory.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
+package org.openstreetmap.atlas.geography.atlas.change;
+
+import org.openstreetmap.atlas.geography.atlas.items.AtlasEntity;
+
+/**
+ * A class that exposes the package private {@link FeatureChange} constructor for testing purposes
+ * only. We want to keep this constructor package private since it is not for general use. However,
```
Comment (matthieun) [inline src/test/java/org/openstreetmap/atlas/geography/atlas/change/FeatureChangeUnitTestFactory.java]: Same comment here: `This constructor package should remain private...`

**Pull request merged**